File: MasterErp.Entities/DTOs/HR/EmployeeAdvanceModel.cs
using MasterErp.Entities.Common.Enums;
using MasterErp.Entities.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MasterErp.Entities.DTOs.HR
{

    public class EmployeeAdvanceModel : CreatorModel
    {
        public int? EmployeeAdvanceId { get; set; }
        public int? EmployeeId { get; set; }
        public string EmployeeName { get; set; }
        public string AdvanceTypeNameEN { get; set; }
        public string AdvanceTypeNameAR { get; set; }
        public int AdvanceTypeId { get; set; }
        public double AdvanceAmount { get; set; }
        public double PaymentAmount { get; set; }
        public DateTime PaymentFromDate { get; set; }
        public DateTime? PaymentToDate { get; set; }
        public string WorkflowStatusNameEN { get; set; }
        public string WorkflowStatusNameAR { get; set; }
        public int? WorkflowStatusId { get; set; } = (int)HRWorkflowStatus.Pending;
        public string Notes { get; set; }
        public double? TotalPaid { get; set; }
        public double? TotalRemaining { get; set; }
        public int? TotalCount { get; set; }


    }
}

File: RetailStation.Angular/src/app/components/HR/models/EmployeeAdvanceModel.ts
import { CreatorModel } from "../../Shared/models/CreatorModel";


export interface EmployeeAdvanceModel extends CreatorModel {
    employeeAdvanceId: number | null;
    employeeId: number | null;
    employeeName: string;
    advanceTypeNameEN: string;
    advanceTypeNameAR: string;
    advanceTypeId: number;
    advanceAmount: number;
    paymentAmount: number;
    paymentFromDate: string;
    paymentToDate: string | null;
    workflowStatusId: number | null;
    workflowStatusNameEN: string;
    workflowStatusNameAR: string;
    isApproved: boolean | null;
    notes: string;
    totalPaid: number | null;
    totalRemaining: number | null;
    totalCount: number | null;
}


export interface AdvancePaymentModel extends CreatorModel {
    advancePaymentId: number | null;
    employeeAdvanceId: number;
    employeeId: number | null;
    employeeName: string;
    moneyAmount: number;
    executionDate: string;
    workflowStatusId: number | null;
    advanceTypeNameEN: string;
    advanceTypeNameAR: string;
    advanceTypeId: number;
    advanceAmount: number;
    workflowStatusNameEN: string;
    workflowStatusNameAR: string;
    notes: string;
    totalCount: number | null;
}